### PR TITLE
Update baselines for snap restore test

### PR DESCRIPTION
**Pull request by andreitraistaru**
## Changes

Update baselines for snap restore test.

## Reason

After finishing the migration to the new infrastructure, some of the baselines got improved. There are also some minor decreases in performance that are correlated with the change in AMIs.

Biggest changes:

- m6a.metal (linux_5.10): target_diff_percentage = -21.96
- m6g.metal (linux_4.14): target_diff_percentage = -5.44
- m5d.metal - 8259CL (linux_5.10): target_diff_percentage = 3.78

Entire comparison output:

```
{
    "source": "3beecc765ddb08727685a7e18bd15bd2f44026e3/tests/integration_tests/performance/configs/",
    "target": "tests/integration_tests/performance/configs/",
    "test_snap_restore_performance_config_4.14.json": {
        "test": "snap_restore_performance",
        "kernel": "4.14",
        "cpus": [
            {
                "instance": "m5d.metal",
                "model": "Intel(R) Xeon(R) Platinum 8175M CPU @ 2.50GHz",
                "stats": {
                    "latency": {
                        "target_diff_percentage": {
                            "mean": -0.6674856266380342,
                            "stdev": 2.235801195407953
                        },
                        "delta_percentage_diff": {
                            "mean": -1.63,
                            "stdev": 7.430016925630218
                        }
                    }
                }
            },
            {
                "instance": "m5d.metal",
                "model": "Intel(R) Xeon(R) Platinum 8259CL CPU @ 2.50GHz",
                "stats": {
                    "latency": {
                        "target_diff_percentage": {
                            "mean": 0.23324895661849318,
                            "stdev": 4.25861989717463
                        },
                        "delta_percentage_diff": {
                            "mean": -9.05,
                            "stdev": 9.433847292991063
                        }
                    }
                }
            },
            {
                "instance": "m6i.metal",
                "model": "Intel(R) Xeon(R) Platinum 8375C CPU @ 2.90GHz",
                "stats": {
                    "latency": {
                        "target_diff_percentage": {
                            "mean": 0.029509416988184923,
                            "stdev": 3.2077475777338176
                        },
                        "delta_percentage_diff": {
                            "mean": 0.18,
                            "stdev": 8.609813522306327
                        }
                    }
                }
            },
            {
                "instance": "m6a.metal",
                "model": "AMD EPYC 7R13 48-Core Processor",
                "stats": {
                    "latency": {
                        "target_diff_percentage": {
                            "mean": 0.8111799179527414,
                            "stdev": 1.584076606868502
                        },
                        "delta_percentage_diff": {
                            "mean": 3.88,
                            "stdev": 18.804297810915898
                        }
                    }
                }
            },
            {
                "instance": "m6g.metal",
                "model": "ARM_NEOVERSE_N1",
                "stats": {
                    "latency": {
                        "target_diff_percentage": {
                            "mean": -5.444182439236914,
                            "stdev": 3.819476518617517
                        },
                        "delta_percentage_diff": {
                            "mean": -4.66,
                            "stdev": 6.782657624002885
                        }
                    }
                }
            },
            {
                "instance": "c7g.metal",
                "model": "ARM_NEOVERSE_V1",
                "stats": {
                    "latency": {
                        "target_diff_percentage": {
                            "mean": 0.7560888806724156,
                            "stdev": 2.285410878339608
                        },
                        "delta_percentage_diff": {
                            "mean": 1.32,
                            "stdev": 3.064557904741749
                        }
                    }
                }
            }
        ]
    },
    "test_snap_restore_performance_config_5.10.json": {
        "test": "snap_restore_performance",
        "kernel": "5.10",
        "cpus": [
            {
                "instance": "m5d.metal",
                "model": "Intel(R) Xeon(R) Platinum 8175M CPU @ 2.50GHz",
                "stats": {
                    "latency": {
                        "target_diff_percentage": {
                            "mean": 1.3229924029264504,
                            "stdev": 1.7684828701290634
                        },
                        "delta_percentage_diff": {
                            "mean": 0.52,
                            "stdev": 5.394497757957274
                        }
                    }
                }
            },
            {
                "instance": "m5d.metal",
                "model": "Intel(R) Xeon(R) Platinum 8259CL CPU @ 2.50GHz",
                "stats": {
                    "latency": {
                        "target_diff_percentage": {
                            "mean": 3.7783081511946004,
                            "stdev": 2.9002233363134122
                        },
                        "delta_percentage_diff": {
                            "mean": -3.2,
                            "stdev": 5.3616671611985325
                        }
                    }
                }
            },
            {
                "instance": "m6i.metal",
                "model": "Intel(R) Xeon(R) Platinum 8375C CPU @ 2.90GHz",
                "stats": {
                    "latency": {
                        "target_diff_percentage": {
                            "mean": -0.05476103772906758,
                            "stdev": 1.4313082326017237
                        },
                        "delta_percentage_diff": {
                            "mean": 1.86,
                            "stdev": 8.707386855916408
                        }
                    }
                }
            },
            {
                "instance": "m6a.metal",
                "model": "AMD EPYC 7R13 48-Core Processor",
                "stats": {
                    "latency": {
                        "target_diff_percentage": {
                            "mean": -21.955185514353474,
                            "stdev": 12.832795930265188
                        },
                        "delta_percentage_diff": {
                            "mean": 7.98,
                            "stdev": 20.961234060363264
                        }
                    }
                }
            },
            {
                "instance": "m6g.metal",
                "model": "ARM_NEOVERSE_N1",
                "stats": {
                    "latency": {
                        "target_diff_percentage": {
                            "mean": 0.5302388236177943,
                            "stdev": 2.0437402650673193
                        },
                        "delta_percentage_diff": {
                            "mean": 1.29,
                            "stdev": 33.47825918986587
                        }
                    }
                }
            },
            {
                "instance": "c7g.metal",
                "model": "ARM_NEOVERSE_V1",
                "stats": {
                    "latency": {
                        "target_diff_percentage": {
                            "mean": -0.15634165102698214,
                            "stdev": 1.4029120870463188
                        },
                        "delta_percentage_diff": {
                            "mean": 4.18,
                            "stdev": 17.160195897259538
                        }
                    }
                }
            }
        ]
    }
}
```

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tools/parse_baselines/main.py [hunk 1]
```diff
@@ -49,7 +59,7 @@ def read_data_files(args):
 
     res_files = [
         f"{filename}_results_{args.instance}_{args.kernel}.ndjson"
-        for filename in OUTPUT_FILENAMES[args.test]
+        for filename in INPUT_FILENAMES[args.test]
```
Comment (pb8o) [inline tools/parse_baselines/main.py:62]: Could be just 

```python
        f"test_{args.test}_results_{args.instance}_{args.kernel}.ndjson"
```

and we wouldn't need the INPUT_FILENAMES dict

File: tools/parse_baselines/main.py [hunk 2]
```diff
@@ -129,7 +139,7 @@ def main():
 
     # Finally, parse and update the baselines.
     baselines_path = Path(
-        f"./tests/integration_tests/performance/configs/test_{args.test}_config_{args.kernel}.json"
+        f"./tests/integration_tests/performance/configs/{OUTPUT_FILENAMES[args.test][0]}_config_{args.kernel}.json"
```
Comment (pb8o) [inline tools/parse_baselines/main.py:142]: We could also just delete the OUTPUT_FILENAMES dict too and depend on the convention that it is always `test_{test_name}...`

**Review by pb8o — DISMISSED** [review]

**Review by zulinx86 — APPROVED** [review]
Looks good to me other than what Pablo commented.

File: tools/parse_baselines/main.py (hunk 2 above)
Comment (andreitraistaru) [reply tools/parse_baselines/main.py]: This is actually the main issue that this PR intends to solve. We currently use `snapshot_restore_performance` as test name, but the output config files that should result from the parser are called `snap_restore_performance`. What I can do to consider your point of view (which I agree with actually) is to keep this convention and also rename those config files from `test_snap_restore_performance_config...` to `test_snapshot_restore_performance_config...`  that we currently have and then we can get rid of both dict you mentioned. However, we should then stick to this convention of naming, but this is not a bad thing overall. Hope I modified everywhere needed not to break smth.

**Review by pb8o — APPROVED** [review]

File: tools/parse_baselines/main.py (hunk 2 above)
Comment (pb8o) [reply tools/parse_baselines/main.py]: aah I see the issue now... `snap` vs `snapshot`. I agree.

**Review by zulinx86 — APPROVED** [review]

**Review by zulinx86 — APPROVED** [review]

**Review by pb8o — APPROVED** [review]

**Pull request merged**